src/libstd/sys/unix/ext/net.rs

```diff
@@ -212,6 +251,20 @@ impl fmt::Debug for UnixStream {
 
 impl UnixStream {
     /// Connects to the socket named by `path`.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::os::unix::net::UnixStream;
+    ///
+    /// let socket = match UnixStream::connect("/tmp/sock") {
```

Comment:
I don't see it as an issue here, on the opposite. The point is to show people how it works, and creating a (unused) variable helps it.